app/javascript/components/supplemental_application/sections/ConfirmedHouseholdIncome.js

```diff
@@ -0,0 +1,137 @@
+import FormGroupTextValue from '~/components/atoms/FormGroupTextValue'
+import { RadioGroup, Radio, Text } from 'react-form';
+import { getAMIAction } from '~/components/supplemental_application/actions'
+
+
+const YesNoRadioGroup = ({field}) => {
+  return (
+    <div className='radio-group-inline'>
+      <RadioGroup field={field}>
+        {(group) => (
+          <React.Fragment>
+            <p className="radio-inline">
+              <Radio group={group} value={`true`} id={`${field}-yes`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-yes`}>Yes</label>
+            </p>
+            <p className="radio-inline">
+              <Radio group={group} value={`false`} id={`${field}-no`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-no`}>No</label>
+            </p>
+          </React.Fragment>
+        )}
+      </RadioGroup>
+    </div>
+  )
+}
+
+const getAMI = ({numHousehold, chartName, chartYear, amis}) => {
+  let ami = find(amis, {'chartType': chartName, 'year': chartYear, 'householdSize': numHousehold})
+
+  if (!isNil(ami)) {
+    return ami.amount
+  }
+}
+
+const getAMIPercent = ({income, ami}) => {
+  if (!income) {
+    return "Enter HH Income"
+  }
+  if (isNil(ami)) {
+    return "Error: AMI unknown"
+  }
+  return income / Number(ami)
+}
+
+
+
+class ConfirmedHouseholdIncome extends React.Component {
+
+  state = {
+    amis: {},
+    amiCharts: []
+  }
+
+  componentDidMount() {
+    // Display a calculated annual income if applicant only provided monthly income
+    if (isNil(this.props.formApi.values.annual_income) && !isNil(this.props.formApi.values.monthly_income)) {
+      let annualIncome = this.props.formApi.values.monthly_income * 12
+      this.props.formApi.setValue('annual_income', annualIncome)
+    }
+
+    // Determine which unique AMI charts are associated with units on the listing
+    let chartsToLoad = uniqBy(this.props.units, (u) => [u.ami_chart_type, u.ami_chart_year].join())
+
+    this.setState({'amiCharts': chartsToLoad.map((chart) => {
+      return {"name": chart.ami_chart_type, "year": chart.ami_chart_year}}
+      )
+    })
+
+    // Load the state with ami values for each chart associated with the listing.
+    chartsToLoad.forEach( (chart) =>
+      getAMIAction({chartType: chart.ami_chart_type, chartYear: chart.ami_chart_year}).then(response => {
+        this.setState({amis: [...this.state.amis, ...response]})
+      }
+    ))
+
+  }
+
+  render() {
+    const { formApi } = this.props
+    const { amis } = this.state
+    return (
+      <React.Fragment>
+      <FormGrid.Row>
+        <FormGrid.Item>
+          <FormGrid.Group label="Recurring Voucher/Subsidy">
+            <YesNoRadioGroup field="housing_voucher_or_subsidy" />
+          </FormGrid.Group>
+        </FormGrid.Item>
+        <FormGrid.Item>
+          <FormGrid.Group label="Household Assets">
+            <Text field="household_assets"
```

Comment:
I do think it can be done without building  custom component. I would strip the $ in the action before calling the the aapi

Is this an option ? we have to ask jesse
![image](https://user-images.githubusercontent.com/188936/44339142-b2e07f80-a456-11e8-8d15-e7a1b22b0a42.png)

Reply:
I think we ended up deciding to move it into another ticket.